### PR TITLE
Fixes #26549 - Registry can handle head requests

### PR DESCRIPTION
**Pull request by parthaa**
This commit adds code to facilitate 'head' requests for information
pertaining to pulling manifests.

### CODE REVIEW

**Comment by parthaa** [comment]
Steps to test:
- Create a docker repo with the following url and upstream name
registry url: https://quay.io/
upstream name: openshift-release-dev/ocp-release
- Sync the repo
- Goto Content->Lifecycle Environments
- Choose library and set unauthenticated pull to yes
-  Try the following call - ```$ curl L --head https://<fqdn>/v2/default_organization<$PRODUCT>-<$REPO>/manifests/4.0.0-0```

Expected results:
```
$ curl -L --head https://theta.partello.example.com/v2/default_organization-docker-busybox/manifests/4.0.0-0
HTTP/1.1 200 OK
Date: Fri, 05 Apr 2019 16:18:31 GMT
Server: Apache
Foreman_version: 1.22.0-develop
Foreman_api_version: 2
Docker-Content-Digest: sha256:59c38273d9331211a5ad2c6fd10634f6db1932c15412e86bbeeba548c9c4c82c
Content-Type: application/json; charset=utf-8
Docker-Distribution-API-Version: registry/2.0
....
```

File: test/controllers/api/registry/registry_proxies_controller_test.rb [hunk 1]
```diff
@@ -267,6 +267,23 @@ def setup_permissions
         assert_equal "dev_label-published_dev_view-puppet_product-busybox", body["results"][0]["name"]
       end
 
+      it "show unauthenticated repositories for head requests" do
```
Comment (mirzal) [inline test/controllers/api/registry/registry_proxies_controller_test.rb]: I kind of feel this test is trying to do too much. Would it make sense to break it down to two tests? One would check that HEAD request works and another would check that Manifest Version 2 Schema 2 correctly rewrites Manifest's `mediaType`.

File: test/controllers/api/registry/registry_proxies_controller_test.rb [hunk 2]
```diff
@@ -329,6 +346,35 @@ def setup_permissions
         assert_equal response.header['Docker-Content-Digest'], "sha256:#{Digest::SHA256.hexdigest(manifest)}"
       end
 
+      it "pull manifest - HTTP Header - v1+json" do
+        #production installs include an HTTPS: 'on' header, which needs to be removed
```
Comment (mirzal) [inline test/controllers/api/registry/registry_proxies_controller_test.rb]: Outdated comment? I don't see this test doing anything about HTTPS header.

File: test/controllers/api/registry/registry_proxies_controller_test.rb (hunk 1 above)
Comment (parthaa) [reply test/controllers/api/registry/registry_proxies_controller_test.rb]: updated. I think the media type is tested else where in other tests in the same file, so decided to leave that out.

**Comment by parthaa** [comment]
[test-katello]

**Comment by parthaa** [comment]
[test_katello]

File: test/controllers/api/registry/registry_proxies_controller_test.rb (hunk 1 above)
Comment (mirzal) [reply test/controllers/api/registry/registry_proxies_controller_test.rb]: Correct, there are other tests here that exercise `mediaType`. Thanks.

**Comment by parthaa** [comment]
[test katello]

**Pull request merged**